src/additional_zones.rs

```diff
@@ -20,15 +20,23 @@ unsafe impl<'a> Send for ZoneWithGeos<'a> {}
 unsafe impl<'a> Sync for ZoneWithGeos<'a> {}
 
 impl<'a> ZoneWithGeos<'a> {
-    fn new(zone: &'a Zone) -> ZoneWithGeos<'a> {
-        ZoneWithGeos {
-            zone,
-            geos: zone
-                .boundary
-                .as_ref()
-                .unwrap()
-                .try_into()
-                .expect("failed to convert to geos"),
+    fn new(zone: &'a Zone) -> Option<ZoneWithGeos<'a>> {
```

Comment:
even if you like `match` I think this can be made a bit shorter with `and_then` :wink: 

something like 

```rust
zone.boundary.as_ref().and_then(b.try_into().map_err(|e| log::error!(gnagnagna e)).ok())
```
? (not tested though, likely to have lots of errors :stuck_out_tongue_winking_eye: